app/concerns/agent_controller_concern.rb

```diff
@@ -68,7 +68,8 @@ def control!
             log "Agent '#{target.name}' is disabled"
           end
         when 'configure'
-          target.update! options: target.options.deep_merge(interpolated['configure_options'])
+          new_options = target.options.deep_merge(interpolated['configure_options']) { |key, old, new| !(new.kind_of?(Array) && old.kind_of?(Array)) ? new : (old+new).uniq }
```

Comment:
There still exists the problem presented when an item in an array is supposed to be changed but not all of the controlled agent's configuration options[within the array] are known by the commander agent. For example, when syncing a profanity filter across multiple Trigger agents that have other rules unique to each agent.

The best solution I've come up with is: within an agents options hash, restricting the use of objects within arrays. 

If an options specification requires an object's worth of data, then it deserves a name(key; unique identifier).